### PR TITLE
Support container args override

### PR DESCRIPTION
**Pull request by michaellzc**
close https://github.com/sourcegraph/sourcegraph/issues/29063

## Test plan

Install a release from the previous commit,

```bash
git checkout HEAD~
helm install -n sourcegraph -f charts/sourcegraph/override.yaml sourcegraph charts/sourcegraph/.
```

Checkout back to the current branch and run `helm diff`, you should see no breaking change

```bash
git checkout 02-22-Support_container_args_override
helm diff -n sourcegraph -f charts/sourcegraph/override.yaml sourcegraph charts/sourcegraph/.
```

### CODE REVIEW

**Comment by michaellzc** [comment]
Current dependencies on/for this PR:
* main
  * **PR #41** <a href="https://app.graphite.dev/github/pr/sourcegraph/deploy-sourcegraph-helm/41" target="_blank">Graphite</a> 
    * **PR #42** <a href="https://app.graphite.dev/github/pr/sourcegraph/deploy-sourcegraph-helm/42" target="_blank">Graphite</a> 
    * **PR #44** <a href="https://app.graphite.dev/github/pr/sourcegraph/deploy-sourcegraph-helm/44" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: charts/sourcegraph/templates/redis/redis-cache.Deployment.yaml [hunk 1]
```diff
@@ -78,6 +82,7 @@ spec:
       - name: redis-exporter
         image: {{ include "sourcegraph.image" (list . "redisExporter") }}
         terminationMessagePolicy: FallbackToLogsOnError
+        env:
         {{- range $name, $item := .Values.redisExporter.env}}
         - name: {{ $name }}
           {{- $item | toYaml | nindent 10 }}
```
Comment (michaellzc) [inline charts/sourcegraph/templates/redis/redis-cache.Deployment.yaml:88]: It was missing the `env:`

**Review by caugustus-sourcegraph — APPROVED** [review]
Nice! A few containers aren't included in this change - was this intentional? None of them seem likely to be customized and most are quite tricky because they're shared across deployments. Just wanted to confirm it was a deliberate choice.

- cadvisor
- jaeger agent (_tracing.tpl)
- alpine init containers
- redis exporter containers
- postgres exporter containers

**Comment by michaellzc** [comment]
> Nice! A few containers aren't included in this change - was this intentional? None of them seem likely to be customized and most are quite tricky because they're shared across deployments. Just wanted to confirm it was a deliberate choice.
> 
> * cadvisor
> * jaeger agent (_tracing.tpl)
> * alpine init containers
> * redis exporter containers
> * postgres exporter containers

I definitely missed advisor

For redis exporter and postgres exporter, we can nest them under the parent services, such as

```yaml
redisCache:
	redisExporter:
      args: ["stuff"]
```

**Pull request merged**